### PR TITLE
[Variant] Make sure variant files are not incorrectly deleted

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

As discussed offline with @sgugger and @ArthurZucker this PR makes sure that only sharded weight files can be deleted to prevent accidentally deleting variants of `pytorch_model.bin` or `model.safetensors` .

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -3119,6 +3119,28 @@ def test_checkpoint_variant_hub_sharded_safe(self):
             )
         self.assertIsNotNone(model)
 
+    @require_safetensors
```
Comment (patrickvonplaten) [inline tests/test_modeling_common.py:3122]:
```suggestion
```

**Comment by patrickvonplaten** [comment]
Failing test is unrelated

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for this fix! 
Should we also have this for `TF`?

**Review by sgugger — APPROVED** [review]
Thanks for fixing and adding a test!

**Comment by patrickvonplaten** [comment]
> this

Might be good to keep in mind as soon as we need "variant" for TF as well

**Pull request merged**